k8s/plugins/openfun/gpgsecretgenerator/GpgSecretGenerator

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+# This Kustomize exec plugins allows to generate a Secret from a GPG encrypted
+# env file (with a symmetric cipher). The passphrase should be available in
+# the SECRET_GPG_PASSWORD environment variable during the execution of this
+# plugin.
+#
+# Example usage:
+#
+# generator.yaml:
+# ---------------
+#
+#  apiVersion: openfun
+#  kind: GpgSecretGenerator
+#  metadata:
+#    name: your-secret-name
+#  argsOneLiner: ./my-file.env.gpg
+
+
+set -eo pipefail
+
+if [[ "$#" -lt 2 ]]; then
+  cat >&2 <<EOF
+    ERROR: You must specify the path to the encrypted env file as an argument
+    Example usage:
+
+    apiVersion: openfun
+    kind: GpgSecretGenerator
+    metadata:
+      name: your-secret-name
+    argsOneLiner: ./my-file.env.gpg
+
+EOF
+  exit 22
+fi
+
+# The first argument is always the path to the YAML configuration file
+# that called this generator
+declare -r plugin_conf=$1
+declare -r encrypted_env_file=$2
+
+# Extract the secret name from the YAML configuration
+secret_name=$(yq e .metadata.name "${plugin_conf}")
+
+if [[ ! -f "${encrypted_env_file}" ]]; then
+  echo "ERROR: Unable to find file ${encrypted_env_file}" >&2
+  exit 2
+fi
+
+if [[ -z "${SECRET_GPG_PASSWORD}" ]]; then
+  echo "You must provide the passphrase to decrypt your env file in the SECRET_GPG_PASSWORD environment variable" >&2
+  exit 22
+fi
+
+# Decrypt the input file
+gpg -q -d --batch --passphrase="${SECRET_GPG_PASSWORD}" "${encrypted_env_file}" | \
```

Comment:
Why not using asymmetric keys ? Public identity is stored in a file, they can be versioned or not, they are public. And you must have the private one when you want to decrypt the secret.

Reply:
I did this because the `gpg` command is executed in a docker container.
And as a user, I would not be comfortable with sharing my GPG secret keyring to any container.

In my idea, the expected "real life" scenario that uses this plugin is as follows:
- A `.team` file containing a list of public keys that are allowed to decrypt the file
- A `passphrase.gpg` file, encrypted using asymetric keys defined in the `.team` file
- The `jitsi-secrets.env` file, encrypted with the passphrase stored in the `passphrase.gpg` file

It's the same principle that we use for [arnold](https://github.com/openfun/arnold/) vaults.

Reply:
:+1: